### DISCUSSION TITLE
code_runner.nvim Edgy Window

### DISCUSSION THREAD

**Discussion by karimlevallois** [LazyVim/LazyVim#2626] [Q&A] [answered]
Hey guys!

Has anyone added code_runner to open in an edgy window rather than the usual split?

**Comment by m0lson84** [comment]
I have not done this with _code_runner_ specifically, but for other plugins I've gotten the filetype of the new window via:

```
:lua print(vim.bo.filetype)
```

Once I know the filetype I'll extend the options for edgy to put it where I want. Here's an example of what I have for `rest-nvim/rest.nvim`:

```
  {
    'folke/edgy.nvim',
    opts = function(_, opts)
      opts.right = vim.list_extend(opts.right or {}, {
        { title = 'REST Client', ft = 'httpResult', size = { width = 0.4 } },
      })
    end,
  },
```

**Reply by karimlevallois** [reply]
It seems to be 'crunner', but still cannot get it to work for some reason. Strange.

```
return {
  {
    "folke/edgy.nvim",
    opts = function(_, opts)
      opts.bottom = vim.list_extend(opts.bottom or {}, {
        { title = "CodeRunner", ft = "crunner", size = { height = 0.4 } },
      })
    end,
  },
}
```

**Reply by m0lson84** [reply, marked as answer]
Looks like the default mode for code_runner.nvim is `term` which will open the output in a new integrated terminal. I was able to get it to work with this example:

```
return {
  {
    'CRAG666/code_runner.nvim',
    opts = {
      mode = 'float',
      filetype = { typescript = 'node run' },
    },
    keys = {
      { '<leader>zz', '<cmd>RunCode<cr>', desc = 'Test running code' },
    },
  },
  {
    'folke/edgy.nvim',
    opts = function(_, opts)
      opts.bottom = vim.list_extend(opts.bottom or {}, {
        { title = 'CodeRunner', ft = 'crunner', size = { height = 0.25 } },
      })
    end,
  },
}
```

**Reply by karimlevallois** [reply]
Awesome, thank you for your help, much appreciated.

**Reply by karimlevallois** [reply]
Sorry to tag you, but as you were so helpful on this issue would you mind helping me on another if you have time. https://github.com/LazyVim/LazyVim/discussions/2635
